plugins/mongo_db/mongo_db_state.cpp

```diff
@@ -239,7 +242,40 @@ namespace mongo_db {
     }
 
     auto state_writer::operator()(const transfer_operation& op) -> result_type {
-        
+        auto oid = std::string(op.from).append("/").append(op.to)
+            .append("/").append(state_block.timestamp);
+        auto oid_hash = fc::sha1::hash(oid).str().substr(0, 24);
```

Comment:
Why do you do it manually? Is it possible to use one function for all cases like `format_oid`?

Reply:
For which case? For "fc::sha1::hash..." - yes, but for "oid = ..." - no, it's different for different object types.

Reply:
The `format_oid` contains expression for calculating oid from string. 
But in this place it seems like same magic.
I request to create a function which calculates hash - and this function should be used in places where it is needed - in format_oid, and in all places where this hash is used for key in bmi.